Support libraries in decomp.me

roblabla opened the pull request:
This MR introduces libraries in decomp.me. This is a new section under "Options" that allow adding libraries to the scratch, which allows the user to `#include` the headers provided by that lib.

Fixes #842

Example:

<img width="967" alt="image" src="https://github.com/decompme/decomp.me/assets/1069318/e844db7c-b905-4470-b481-6b1f2ea7caf5">


This is ready for review.

roblabla commented:
There's one mypy error that I just can't figure out for the life of me. If someone can find a fix, I'd be grateful.

Otherwise, this feature is ready for review.

mkst commented on backend/compilers/download.py:
```diff
@@ -237,6 +238,133 @@ def get_compiler(
     return True
 
 
+def get_library(
```
I feel like the library downloading should be a separate script from the compilers downloading. Dunno what other people's opinions are though.

ethteck replied:
ah, yeah that didn't really register to me, but I agree

roblabla replied:
Yeah that's fair. It used to reuse a lot of the infrastructure from compilers/download.py so it made sense to have a single script, but now that it does its own stuff, it can move to its own script.

roblabla replied:
Fixed. I also updated the docs to explain what the new libraries folder does.

nanaian reviewed (approved):
Frontend changes lgtm

ethteck commented:
<img width="621" alt="image" src="https://github.com/decompme/decomp.me/assets/2985314/0b9b87d9-6c0f-4ec5-b10e-e74933d087b6">
I think the library flag(s) that get appended should always start with a space

roblabla commented:
> I think the library flag(s) that get appended should always start with a space

Nice catch, I added a space, and also moved the libraries to be within `sandbox.quote_options`
